helix-term/src/ui/popup.rs

```diff
@@ -207,6 +243,67 @@ impl<T: Component> Component for Popup<T> {
                 self.scroll(self.size.1 as usize / 2, false);
                 EventResult::Consumed(None)
             }
+            ctrl!('o') => {
+                let (stringified_body, stringified_buffername, stringified_language) =
+                    if let Some(contents_as_strings) = &self.contents_as_strings {
+                        let stringified_body = contents_as_strings.body.clone();
+                        let stringified_buffername = contents_as_strings.title.clone();
+                        let stringified_language = contents_as_strings.language.clone();
+                        (
+                            stringified_body,
+                            stringified_buffername,
+                            stringified_language,
+                        )
+                    } else {
+                        ("dummy".into(), self.id.into(), "none".into())
+                    };
+
+                // open a new buffer to show the popup content
+                cx.editor.new_file(Action::Replace);
+
+                // TODO: how can we meaningfully stringify the content of a Popup? for now dummy
+                // TODO: is this an un-necessary copy? ...
+                // put the content in the new buffer
+                let (view, doc) = current!(cx.editor);
+                paste_impl(
+                    &[stringified_body],
+                    doc,
+                    view,
+                    Paste::Before,
+                    1,
+                    cx.editor.mode,
+                );
```

Comment:
The new document should be created with `Document::from` and then added to the editor with `Editor::new_file_from_document` (which is currently private but could be `pub`). That will remove the need to modify the selection, jumplist and last-saved-revision.